protocol/communities/permission_checker.go

```diff
@@ -105,6 +112,50 @@ func (p *DefaultPermissionChecker) GetOwnedERC721Tokens(walletAddresses []gethco
 	return ownedERC721Tokens, nil
 }
 
+func (p *DefaultPermissionChecker) GetOwnedERC721Tokens(walletAddresses []gethcommon.Address, tokenRequirements map[uint64]map[string]*protobuf.TokenCriteria, chainIDs []uint64) (CollectiblesByChain, error) {
+	if p.collectiblesManager == nil {
+		return nil, errors.New("no collectibles manager")
+	}
+
+	ctx := context.Background()
+
+	query := ERC721TokenQuery{
+		WalletAddresses:   walletAddresses,
+		TokenRequirements: tokenRequirements,
+		ChainIDs:          chainIDs,
+	}
+
+	return fetchOwnedERC721Tokens(
+		ctx,
+		p.collectiblesManager,
+		query,
+		p.collectiblesManager.FetchBalancesByOwnerAndContractAddress,
+		p.logger,
+	)
+}
+
+func (p *DefaultPermissionChecker) GetOwnedERC721TokensFromCache(walletAddresses []gethcommon.Address, tokenRequirements map[uint64]map[string]*protobuf.TokenCriteria, chainIDs []uint64) (CollectiblesByChain, error) {
+	if p.collectiblesManager == nil {
+		return nil, errors.New("no collectibles manager")
+	}
+
+	ctx := context.Background()
+
+	query := ERC721TokenQuery{
+		WalletAddresses:   walletAddresses,
+		TokenRequirements: tokenRequirements,
+		ChainIDs:          chainIDs,
+	}
```

Comment:
Since all of these lines are the same for both functions, could we move them to `fetchOwnedERC721Tokens`?